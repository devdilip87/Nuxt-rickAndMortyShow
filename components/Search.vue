<template>
    <div class="search">
        <input type="text" placeholder="Search by Name" v-model="name"/>
        <!--<button @click="onSearch">Search</button>-->
    </div>
</template>

<script>
import SelectedFilterVue from './SelectedFilter.vue';
    export default {
        data () {
            return {
                name: ''
            }
        },
        watch:{
            name: function(n,o){
                const payload = {
                    name: this.name,
                    list: this.$store.state.characters,
                    selectedFilter: this.$store.state.selectedFilterOptions
                }
                this.$store.dispatch("nameSearch", payload);
            }
        }
    }
</script>

<style>
</style>
