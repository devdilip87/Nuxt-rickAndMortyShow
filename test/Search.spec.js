import { mount, shallowMount } from '@vue/test-utils'
import Search from '@/components/Search.vue'

describe('Search', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Search, {
            data: () => {
                return {
                    name: ''
                };
            }
        })
    });

    test('generate snapshot test', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test("to check name prop for null condition", () => {
        wrapper.vm.name = null;
        expect(wrapper.vm.name).toBeNull();
    })

    test("to check name prop falsy condtion on watch", () => {
        wrapper.vm.name = '';
        expect(wrapper.vm.name).toBeFalsy();
    })

    test("to check name prop truthy condition on watch", () => {
        wrapper.vm.name = 'Dilip';
        expect(wrapper.vm.name).toBeTruthy();
    })

    test("check `name` prop change prooperly on watch", () => {
        wrapper.vm.name = 'Alan';
        expect(wrapper.vm.name).toStrictEqual('Alan');
    })
})