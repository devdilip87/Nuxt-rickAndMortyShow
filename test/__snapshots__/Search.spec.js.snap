// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search generate snapshot test 1`] = `"<div class=\\"search\\"><i class=\\"fa fa-search\\"></i> <input type=\\"text\\" placeholder=\\"Search by Name\\"></div>"`;
